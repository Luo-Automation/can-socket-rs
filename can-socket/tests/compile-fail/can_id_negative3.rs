fn main() {
	let _id = can_socket::id!(extended: -2);
}
