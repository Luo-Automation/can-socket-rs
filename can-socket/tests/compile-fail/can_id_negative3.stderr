error[E0277]: the trait bound `u32: Neg` is not satisfied
 --> tests/compile-fail/can_id_negative3.rs:2:38
  |
2 |     let _id = can_socket::id!(extended: -2);
  |                                         ^^ the trait `Neg` is not implemented for `u32`
  |
  = help: the following other types implement trait `Neg`:
            &f128
            &f16
            &f32
            &f64
            &i128
            &i16
            &i32
            &i64
          and $N others

error[E0600]: cannot apply unary operator `-` to type `u32`
 --> tests/compile-fail/can_id_negative3.rs:2:38
  |
2 |     let _id = can_socket::id!(extended: -2);
  |                                         ^^ cannot apply unary operator `-`
  |
  = note: unsigned values cannot be negated
